### ISSUE TITLE
deselectAllDates() 

### ISSUE THREAD

**Issue by Distance-MJ** [patchthecode/JTAppleCalendar#1184]
when I use deselectAllDates()  function with v7.1.0, not  perform  calendar(_ calendar: JTAppleCalendarView, shouldDeselectDate date: Date, cell: JTAppleCell?, cellState: CellState)

**Comment by patchthecode** [comment]
why you do not want it to call `shouldSelect`?

**Comment by Distance-MJ** [comment]
I setup triggerSelectionDelegate == false, I think it should not call  shouldSelect
shouldSelect function has a callback, resulting in cyclic calls
v7.1.0 not support Swift 5 regrettably
This problem only appears after the latest update.

**Comment by Distance-MJ** [comment]
> why you do not want it to call `shouldSelect`?

I setup triggerSelectionDelegate == false, I think it should not call shouldSelect
shouldSelect function has a callback, resulting in cyclic calls
v7.1.0 not support Swift 5 regrettably
This problem only appears after the latest update.

**Comment by patchthecode** [comment]
is it fixed in version 7.1.8?

**Comment by patchthecode** [comment]
If not I will look at this over the weekend.
I have a number of issues to fix that I have put on the desk for a while.

**Comment by Distance-MJ** [comment]
> If not I will look at this over the weekend.
> I have a number of issues to fix that I have put on the desk for a while.

yes,  I found v7.1.6, v7.1.7,v8.0.0+ which support Swift 5's version all has this problem.
thank.

**Comment by patchthecode** [comment]
OK, will look at it over the weekend.
The fix should be simple.
--> If user has `triggerSelectionDelegate`  set to false then i will not trigger the should select

**Comment by Distance-MJ** [comment]
> OK, will look at it over the weekend.
> The fix should be simple.
> --> If user has `triggerSelectionDelegate` set to false then i will not trigger the should select

ok, thank you.
when can I update it?
